src/cart/src/swim/swim.c

```diff
@@ -278,14 +278,25 @@ swim_member_alive(struct swim_context *ctx, swim_id_t from,
 	/* if member is suspected, remove from suspect list */
 	TAILQ_FOREACH(item, &ctx->sc_suspects, si_link) {
 		if (item->si_id == id) {
+			uint64_t timeout = swim_ping_timeout_get();
+
 			/* remove this member from suspect list */
 			TAILQ_REMOVE(&ctx->sc_suspects, item, si_link);
-			if (swim_ping_timeout_get() < swim_period_get()) {
-				swim_ping_timeout_set(swim_ping_timeout_get() +
-						      SWIM_PING_TIMEOUT);
-				SWIM_INFO("%lu: increase ping timeout to %lu\n",
-					 ctx->sc_self, swim_ping_timeout_get());
+
+			timeout += timeout / 2;
```

Comment:
probably need to have an upper bound when increase it, or it possibly cause ping timeout be increased to a un-reasonable value?  And seems the increase step - 1.5x each time seems too aggressive?
or maybe just one place (in crt_swim_srv_cb as above) to tune the parameter is enough (as here SUSPECT->ALIVE change is tolerable in SWIM).

Reply:
I agree it should be an upper bound. But what bound is reasonable enough? I don't have clear vision about this right now. And I don't want to introduce more hardcoded parameters.

From my point of view it's not aggressive. It should be updated once or twice, from my expectation. Then, if network will be stable enough, they will not be updated. According to my experiments it's not increased more because of average value can not significantly increased for few short network glitches.

Maybe better approach here will be update average from more items.